### PR TITLE
Use style guide checkbox and allow click on column to select row.

### PR DESCRIPTION
**Pull request by xerona**
@caffeinelogic @akersinformz 
https://jira.inbcu.com/browse/AUG-181 still needs implementation in audri
resolves https://github.com/CINBCUniversal/suprematism-data-grid/issues/24

### CODE REVIEW

File: dist/data-grid-column/data-grid-column.component.html [hunk 1]
```diff
@@ -33,7 +35,12 @@
         <span class="js-lockedColumn lockColumn">Column Locked <span class="ion-locked"></span></span>
         <span class="js-unlockedColumn lockColumn">Column Unlocked <span class="ion-unlocked"></span></span>
       </div>
-      <ng-content></ng-content>
+      <div *ngIf="checksCheckbox" supreCheckBoxSync>
+        {{content}}
```
Comment (caffeinelogic) [inline dist/data-grid-column/data-grid-column.component.html]: Why is this not using `ng-content`? Just curious.

File: src/data-grid/data-grid-column/data-grid-column.component.spec.ts [hunk 2]
```diff
@@ -10,7 +11,8 @@ describe('DataGridColumn', () => {
     TestBed.configureTestingModule({
       declarations: [
         DataGridColumnComponent
-      ]
+      ],
+      providers: [StateManagerService]
```
Comment (caffeinelogic) [inline src/data-grid/data-grid-column/data-grid-column.component.spec.ts]: Do we need this?

File: src/data-grid/data-grid-row/data-grid-row.component.ts [hunk 3]
```diff
@@ -145,23 +145,21 @@ export class DataGridRowComponent implements AfterContentInit {
...
-    if (element.classList.contains('icon')) {
-      if (element.classList.contains('ion-android-arrow-dropright-circle')) {
-        this.setRowOpenState(element, (<HTMLDivElement>supreDataGridRow.firstElementChild), true);
-        this.triggerEvent(supreDataGridRow, 'add');
-      } else {
-        this.setRowOpenState(element, (<HTMLDivElement>supreDataGridRow.firstElementChild), false);
-        this.triggerEvent(supreDataGridRow, 'minus');
-        this.checkChildRowVisibilityService.fireEvent();
-      }
+    if (element.classList.contains('ion-android-arrow-dropright-circle')) {
```
Comment (caffeinelogic) [inline src/data-grid/data-grid-row/data-grid-row.component.ts]: We need to replace these icons at somepoint, should be using the styleguide fonts. Not part of this story though.

@akersinformz @xerona

File: src/data-grid/shared/directives/checkbox-sync.directive.ts [hunk 4]
```diff
@@ -13,19 +13,19 @@ type checkBoxEvent = {
...
-}
+};
 
 @Directive({
     selector: '[supreCheckBoxSync]'
 })
 export class CheckBoxSyncDirective implements AfterViewInit, OnDestroy {
     private observablesToUnsubscribe: ISubscription[] = [];
-    @HostBinding('value') value: string;
+    @Input() value: string;
```
Comment (caffeinelogic) [inline src/data-grid/shared/directives/checkbox-sync.directive.ts]: @akersinformz 
Is this correct. @xerona and I worked on this and came up with this.

File: dist/data-grid-column/data-grid-column.component.html (hunk 1 above)
Comment (xerona) [inline dist/data-grid-column/data-grid-column.component.html]: I don't know what was happening here. I exceeded my timebox trying to figure it out, but the ng-content was being dropped.

File: src/data-grid/shared/directives/checkbox-sync.directive.ts (hunk 4 above)
Comment (akersnbc) [inline src/data-grid/shared/directives/checkbox-sync.directive.ts]: Yeah that will work

Not sure why it was a @HostBinding('value')

**Pull request merged**